tests/Integration/Behaviour/Features/Scenario/Product/update_stock_advanced.feature

```diff
@@ -0,0 +1,292 @@
+      | is_virtual | true        |
+    Then product "product1" should have following stock information:
+      | out_of_stock_type | available |
+
+  Scenario: I update product quantity
+    Given I add product "product1" with following information:
+      | name       | en-US:Presta camera |
+      | is_virtual | false               |
+    And product "product1" should have following stock information:
+      | quantity | 0 |
+    When I update product "product1" stock with following information:
+      | quantity | 51 |
+    Then product "product1" should have following stock information:
+      | quantity | 51 |
+    And product "product1" last stock movement has following details:
+      | physical_quantity | 51 |
+      | sign              | 1  |
+    When I update product "product1" stock with following information:
+      | quantity | 42 |
+    Then product "product1" should have following stock information:
+      | quantity | 42 |
+    And product "product1" last stock movement has following details:
+      | physical_quantity | 9  |
+      | sign              | -1 |
+
+  Scenario: I update product quantity specifying if movement must be added or not
+    Given I add product "product1" with following information:
+      | name       | en-US:Presta camera |
+      | is_virtual | false               |
+    And product "product1" should have following stock information:
+      | quantity     | 0 |
+    When I update product "product1" stock with following information:
+      | quantity     | 51    |
+      | add_movement | false |
+    And product "product1" has no stock movements
+    When I update product "product1" stock with following information:
+      | quantity     | 42   |
+      | add_movement | true |
+    Then product "product1" should have following stock information:
+      | quantity | 42 |
+    And product "product1" last stock movement has following details:
+      | physical_quantity | 9  |
+      | sign              | -1 |
+
+  Scenario: I update product simple stock fields
+    Given I add product "product1" with following information:
+      | name       | en-US:Presta camera |
+      | is_virtual | false               |
+    And product "product1" should have following stock information:
+      | minimal_quantity       | 1          |
+      | location               |            |
+      | low_stock_threshold    | 0          |
+      | low_stock_alert        | false      |
+      | available_date         | 0000-00-00 |
+    And product "product1" localized "available_now_labels" should be "en-US:"
+    And product "product1" localized "available_later_labels" should be "en-US:"
+    When I update product "product1" stock with following information:
+      | minimal_quantity       | 12                 |
+      | location               | dtc                |
+      | low_stock_threshold    | 42                 |
+      | low_stock_alert        | true               |
+      | available_now_labels   | en-US:get it now   |
```

Comment:
you might want to check multiple langs at some point :smile: 
(at least a scenario to check if required localized fields are filled with default lang value when you aren't providing them)

Reply:
Yep nice catch!

Reply:
I added checked with french language it works correctly when it is provided However there is no fallback system on the default language I'm not sure we have implemented this earlier because I believe all other tests always provide the two values And I'm not even sure it has ever been handled in our commands 🤔  Isn't it handled by the form constraint usually? or the front js which automatically populates missing fields?

Do you have any example of fallback?

Reply:
I think as we discussed its already resolved?
for the record - only required properties are filled with default value, all other are filled with empty string if not provided at all.